dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/metrics/ChartTableCreator.java

```diff
@@ -0,0 +1,46 @@
+package edu.harvard.iq.dataverse.metrics;
+
+import edu.harvard.iq.dataverse.common.BundleUtil;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.primefaces.model.chart.AxisType;
+import org.primefaces.model.chart.BarChartModel;
+
+import javax.ejb.Stateless;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Stateless
+public class ChartTableCreator {
+
+    public ChartTableModel createChartTable(BarChartModel barChartModel) {
+        ChartTableModel tableModel = new ChartTableModel();
+        tableModel.setDataRow(loadDataForChartTable(barChartModel));
+        tableModel.setTitle(barChartModel.getTitle());
+        tableModel.setLeftColumnName(barChartModel.getAxis(AxisType.X).getLabel());
+        tableModel.setRightColumnName(barChartModel.getAxis(AxisType.Y).getLabel());
+
+        return tableModel;
+    }
+
+    public ChartTableModel createMonthlyChartTable(BarChartModel barChartModel, int year) {
+        ChartTableModel tableModel = new ChartTableModel();
+        tableModel.setDataRow(loadDataForChartTable(barChartModel));
+        tableModel.setTitle(getMonthlyChartTableTitle(barChartModel, year));
+        tableModel.setLeftColumnName(barChartModel.getAxis(AxisType.X).getLabel());
+        tableModel.setRightColumnName(barChartModel.getAxis(AxisType.Y).getLabel());
+
+        return tableModel;
+    }
+
+    private List<Pair<String, String>> loadDataForChartTable(BarChartModel chartModel) {
+        return chartModel.getSeries().get(0).getData().entrySet().stream()
```

Comment:
Is this chain null-safe in every case?

Reply:
For empty charts (with no backing data) an empty data row for current year is being created anyhow. That translates into `chartModel.getSeries` always having at least one element so it should be null-safe.
However this detail is specific to the way we are creating charts  so maybe a check is in place? What do you think?

Reply:
Yes, if it can somehow be misused later, then it's better to be prepared: maybe some kind of Optional or something similar.